gms/gossiper.hh

```diff
@@ -85,6 +86,11 @@ struct syn_msg_pending {
     std::optional<gossip_digest_syn> syn_msg;
 };
 
+struct ack_msg_pending {
+    bool pending = false;
```

Comment:
`pending` seems redundant, could we use `bool(ack_msg_digest) `?

Reply:
It is not redundant. When we consume p.ack_msg_digest and send the ack2 msg, p.pending is still true, but bool(p.ack_msg_digest) is false.

Reply:
Right